configs/imxrt1050-evk/loadable_elf_apps : Generate stripped ELF

kishore-sn opened the pull request:
Add -s flag to linker flags to generated stripped ELF binary

seinfra commented:
Target : [ea833a4240c031fb7ec497ec63932c767945cfbd] - Code Rule Check (C++) OK.

seinfra commented:
Target : [ea833a4240c031fb7ec497ec63932c767945cfbd] - Code Rule Check OK.

sunghan-chang commented:
@kishore-sn Could you apply this like below?
```
ifeq ($(CONFIG_DEBUG_SYMBOLS),y)
  LDFLAGS += -g
endif
```

sunghan-chang commented:
And could you add that at all of Make.defs under configs?

btheosam commented on build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs:
```diff
@@ -139,7 +139,7 @@ AFLAGS = $(CFLAGS) -D__ASSEMBLY__
 CXXELFFLAGS = $(CXXFLAGS) -mlong-calls # --target1-abs
 
-LDELFFLAGS = -r -e main
+LDELFFLAGS = -r -e main -s
```
Better to put inside a flag and if dynamic linking is enabled, then this flag should n't be there. Account  these.

kishore-sn commented:
I made the below changes:
- Check for CONFIG_DEBUG_SYMBOLS and add -s option
- Disable CONFIG_DEBUG_SYMBOLS in defconfig

I have made these changes only in loadable_elf_apps config because LDELFFLAGS is defined only in this config.

seinfra commented:
Target : [ea303a9683de21fb4809d651b117faaae730de00] - Code Rule Check OK.

seinfra commented:
Target : [ea303a9683de21fb4809d651b117faaae730de00] - Code Rule Check (C++) OK.

sunghan-chang commented:
@kishore-sn What I wrote is an example. Is it appropriatary with CONFIG_DEBUG_SYMBOLS?

kishore-sn commented:
Ok. I understand. 
I added a new config for this.

seinfra commented:
Target : [9e8ade1a9aa0e646fc85bc41641e3dd6e505d42f] - Code Rule Check OK.

seinfra commented:
Target : [9e8ade1a9aa0e646fc85bc41641e3dd6e505d42f] - Code Rule Check (C++) OK.

sunghan-chang commented on os/binfmt/libelf/Kconfig:
```diff
@@ -48,3 +48,10 @@ config ELF_EXIDX_SECTNAME
 
 		This is needed to support exception handling on loadable ELF modules.
+
+config ELF_INCLUDE_SYMBOLS
```
1. How about `exclude` and `ifeq ($(CONFIG_ELF_EXCLUDE_SYMBOLS),y)`?
`include` looks positive but real operation `ifneq` is negative.
I think exclude is better to understand.
2.  Is this proper location to add? Here is not a location of making binary. Isn't it?
3. Symbol information is necessary to debug using T32. Then, how about making a new binary which is stripped?
I mean after making non-stripped elf and then making new stripped file to download.

sunghan-chang replied:
for 3rd, Cypress does like what I said.

sunghan-chang commented on build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs:
```diff
@@ -140,6 +140,9 @@ CELFFLAGS = $(CFLAGS) -mlong-calls # --target1-abs
 CXXELFFLAGS = $(CXXFLAGS) -mlong-calls # --target1-abs
 
 LDELFFLAGS = -r -e main
+ifneq ($(CONFIG_ELF_INCLUDE_SYMBOLS),y)
```
Could you add this at every Make.defs?

sunghan-chang replied:
I saw your comment now. Please ignore this.

sunghan-chang commented on build/configs/imxrt1050-evk/loadable_elf_apps/defconfig:
```diff
@@ -823,6 +822,7 @@ CONFIG_ELF_ALIGN_LOG2=2
 CONFIG_ELF_STACKSIZE=2048
 CONFIG_ELF_BUFFERSIZE=32
 CONFIG_ELF_BUFFERINCR=32
+# CONFIG_ELF_INCLUDE_SYMBOLS is not set
```
You must keep a file permission with 644. Please revert

kishore-sn commented:
@sunghan-chang  I made below changes
1. Changed the config to CONFIG_ELF_EXCLUDE_SYMBOLS
2. Moved the logic from Make.defs to Makefile
3. Added logic to keep both stripped and non-stripped binary files
4. The non stripped file will be named as micom_dbg  and the stripped binary is named as micom
5. Changed file permission

seinfra commented:
Target : [d9f4a24539193fac81f1d6d8f798ecae74ee2fc1] - Code Rule Check OK.

seinfra commented:
Target : [d9f4a24539193fac81f1d6d8f798ecae74ee2fc1] - Code Rule Check (C++) OK.

sijupunnnoose commented on build/configs/imxrt1050-evk/loadable_elf_apps/defconfig:
```diff
@@ -285,7 +285,6 @@ CONFIG_ARCH_HAVE_BUTTONS=y
 # CONFIG_ARCH_BUTTONS is not set
```
100755->100644

sunghan-chang commented on apps/examples/elf/micomapp/Makefile:
```diff
@@ -113,6 +113,10 @@ clean:
 install:
 	$(Q) mkdir -p $(USER_BIN_DIR)
 	$(Q) install $(BIN) $(USER_BIN_DIR)/$(BIN)
+ifeq ($(CONFIG_ELF_EXCLUDE_SYMBOLS),y)
```
I suggest two location to add this.
1. Makefile.unix
```
...
ifeq ($(CONFIG_RAW_BINARY),y)
	$(Q) echo "CP: $(BIN_EXE).bin"
	$(Q) $(OBJCOPY) $(OBJCOPYARGS) -O binary $(BIN) $(BIN).bin
endif
HERE!!!!!!!!!!!!!!!!!!!!!!!!!
	$(Q) echo "Start the Board Specific Work for Binary"
	$(Q) $(call MAKE_BOARD_SPECIFIC_BIN, $(BIN_EXE), $(BIN_EXT))
...
```
2. In MAKE_BOARD_SPECIFIC_BIN definition of each board.

kishore-sn replied:
We have below line in the micomapp/Makefile and wifiapp/Makefile

`$(Q) $(TOPDIR)/tools/mkbinheader.py $(USER_BIN_DIR)/$(BIN) $(BIN_TYPE) $(KERNEL_VER) $(BIN) $(BIN_VER) $(DYNAMIC_RAM_SIZE)`

It is not possible to apply the strip command after adding the header. So, I need to make these changes in these Makefile's only.

seinfra commented:
Target : [2d3b38d3ad5ad1cf618f0e4add32d9cc82f1d40a] - Code Rule Check (C++) OK.

seinfra commented:
Target : [2d3b38d3ad5ad1cf618f0e4add32d9cc82f1d40a] - Code Rule Check OK.
